docs/source/design/dataset_argument.rst

```diff
@@ -0,0 +1,63 @@
+.. -*- mode: rst -*-
+.. vi: set ft=rst sts=4 ts=4 sw=4 et tw=79:
+
+.. _chap_design_dataset_argument:
+
+********************
+``dataset`` argument
+********************
+
+.. topic:: Specification scope and status
+
+   This specification describes the current implementation.
+
+All commands which operate on datasets have a ``dataset`` argument (``-d`` or
+``--dataset`` for the :term:`CLI`) to identify a single dataset as the
+If ``--dataset`` argument is not provided, the context of an operation is operation-specific (e.g., for `clone` command -- the dataset which is being cloned), but typically - a dataset which current working directory belongs to.
+In the latter case, if operation (e.g., `get`) does not find a dataset in current working directory, operation fails with an ``NoDatasetFound`` error.
```

Comment:
if we decide to not "migrate" toward a sentence per line in written text, I will wrap them.
WDYT @mih @bpoldrack ?